pydmd/costs.py

```diff
@@ -0,0 +1,1457 @@
+        :rtype: slice
+        """
+        # Get the window indices and data.
+        sample_start = self._step_size * k
+        if k == self._n_slides - 1 and self._non_integer_n_slide:
+            sample_slice = slice(-self._window_length, None)
+        else:
+            sample_slice = slice(
+                sample_start, sample_start + self._window_length
+            )
+        return sample_slice
+
+    def cluster_omega(
+        self,
+        n_components,
+        kmeans_kwargs=None,
+        transform_method=None,
+        method="KMeans",
+    ):
+        """Clusters fitted eigenvalues into frequency bands by the imaginary component.
+
+        :param method: Choose clustering strategy: KMeans (default) or KMediods
+            (requires sklearn_extras).
+        :type method: str
+        :param n_components: Hyperparameter for k-means clustering, number of clusters.
+        :type n_components: int
+        :param kmeans_kwargs: Arguments for KMeans clustering.
+        :type kmeans_kwargs: dict
+        :param transform_method: How to transform omega. See docstring for valid options.
+        :type transform_method: str or NoneType
+        :return:
+        """
+        # Reshape the omega array into a 1d array
+        omega_array = self.omega_array
+        n_slides = omega_array.shape[0]
+        svd_rank = omega_array.shape[1]
+        omega_rshp = omega_array.reshape(n_slides * svd_rank)
+        omega_transform = self.transform_omega(
+            omega_rshp, transform_method=transform_method
+        )
+
+        if kmeans_kwargs is None:
+            random_state = 0
+            kmeans_kwargs = {
+                "random_state": random_state,
+            }
+        if method == "KMeans":
+            clustering = KMeans(n_clusters=n_components, **kmeans_kwargs)
+        elif method == "KMediods":
+            from sklearn_extra.cluster import KMedoids
+
+            clustering = KMedoids(n_clusters=n_components, **kmeans_kwargs)
+        else:
+            raise ValueError(
+                "Unrecognized clustering method {}.".format(method)
+            )
+
+        omega_classes = clustering.fit_predict(np.atleast_2d(omega_transform).T)
+        omega_classes = omega_classes.reshape(n_slides, svd_rank)
+        cluster_centroids = clustering.cluster_centers_.flatten()
+
+        # Sort the clusters by the centroid magnitude.
+        idx = np.argsort(cluster_centroids)
+        lut = np.zeros_like(idx)
+        lut[idx] = np.arange(n_components)
+        omega_classes = lut[omega_classes]
+        cluster_centroids = cluster_centroids[idx]
+
+        # Assign the results to the object.
+        self._cluster_centroids = cluster_centroids
+        self._omega_classes = omega_classes
+        self._transform_method = transform_method
+        self._n_components = n_components
+
+        return self
+
+    def transform_omega(self, omega_array, transform_method=None):
```

Comment:
```suggestion
    def transform_omega(self, omega_array, transform_method="absolute"):
```
And then we should throw an exception if `transform_method` is `None`